source/utilities/http.ts

```diff
@@ -0,0 +1,88 @@
+// source/utilities/http.ts
+// Helper functions for the server.
+
+import { parse } from 'node:url';
+import { networkInterfaces as getNetworkInterfaces } from 'node:os';
+import type { ListenEndpoint, ParsedEndpoint } from '../types.js';
+
+const networkInterfaces = getNetworkInterfaces();
+
+/**
+ * Parse and return the endpoints from the given string.
+ *
+ * @param uriOrPort - The endpoint to listen on.
+ * @returns A list of parsed endpoints.
+ */
+export const parseEndpoint = (uriOrPort: ListenEndpoint): ParsedEndpoint => {
+  // If the endpoint is a port number, return it as is.
+  // @ts-expect-error `isNaN` accepts strings too.
+  if (!isNaN(uriOrPort)) return [uriOrPort];
```

Comment:
When you 
```suggestion
  if (!isNaN(Number(uriOrPort))) return [uriOrPort];
```

This then surfaces another issue which is why we ban `// @ts-ignore`. Can you please fix this in both places?

Reply:
Fixed it ;]